client/components/button-fill.vue

```diff
@@ -0,0 +1,50 @@
+<template>
+  <component
+    class="button-fill"
+    :href="href"
+    :is="tag"
```

Comment:
Do you really need this to be this general (with prop defining type of HTML element)?

If you do, then I think you might be missing additional style resets below.

Reply:
My view on this is a button should look like a button, regardless of semantics. i.e. in the case for archival screen the button changes the url from /basic to /advanced. Another use case might be a confirm button on a modal. Both buttons should still look like buttons even if one is a button type=submit and one is using a router-link.

I'm happy to add any style resets that you can think of including. I was thinking to add any i've missed when the use case comes up for it.

Reply:
If you are trying to support button and anchor tag, then look at the browser defaults for those. For example, links typically have `text-decoration: underline` on hover.

The discussion of "a button should look like a button regardless of semantics" depends on the design. Not saying this styling isn't valid, however how this is setup in code can be accomplished in different ways and depending on a number of variables it might make sense to split them out semantically and share styles across each. Not sure how this would look in your project, but just wanted to call it out for consideration.